dockerfiles/base-ci-linux/Dockerfile

```diff
@@ -20,24 +20,32 @@ dockerfiles/rust-builder/README.md" \
 
 WORKDIR /builds
 
-# install tools and dependencies
-COPY clang9.key /etc/apt/trusted.gpg.d/debian-archive-jessie-automatic.gpg
 # config for wasm32-unknown-unknown
-COPY config /root/.cargo/config
+COPY utility/base-ci-linux-config /root/.cargo/config
+
 # gcc and clang are necessary to be able to compile crates that have build scripts
 # Most notably, clang 9 has WASM support enabled by default.
+COPY utility/debian-llvm-clang9.key /etc/apt/trusted.gpg.d/debian-archive-llvm.gpg
+
+# copy cmake
+COPY utility/cmake.sh cmake.sh
+
+# install tools and dependencies
 RUN set -eux; \
-	echo "deb http://apt.llvm.org/stretch/ llvm-toolchain-stretch-9 main" \
+	echo "deb http://apt.llvm.org/buster/ llvm-toolchain-buster-9 main" \
 		> /etc/apt/sources.list.d/llvm.list; \
-	echo "deb-src http://apt.llvm.org/stretch/ llvm-toolchain-stretch-9 main" \
+	echo "deb-src http://apt.llvm.org/buster/ llvm-toolchain-buster-9 main" \
 		>> /etc/apt/sources.list.d/llvm.list; \
-	DEBIAN_FRONTEND=noninteractive apt-get -y update; \
-	DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
-		g++ libssl-dev gcc clang-9 cmake make \
+	apt-get -y update; \
+	apt-get install -y --no-install-recommends \
+		g++ libssl-dev gcc clang-9 libclang-9-dev make \
 		git pkg-config curl time rhash; \
 # set a link to clang-9
 	ln -s /usr/bin/clang-9 /usr/bin/clang; \
+# install cmake
+	./cmake.sh --skip-license --prefix=/usr/ --exclude-subdir; \
```

Comment:
IMO it makes no sense to rebuild the same archive-containing script again every night. The result won't change.

Reply:
what do you propose here?

Reply:
use a precompiled debian package like https://packages.debian.org/bullseye/cmake
cmake patch release `3.16.2` is out and it will not be included here (instead the same outdated version is build over and over again). If it's really a big concern about using the same version everywhere, why not use this one - makes things way easier.

Reply:
that definitely makes sense, mill mark myself to do that later. Thanks!